src/parsimmon.js

```diff
@@ -394,44 +405,56 @@ function formatGot(input, error) {
   if (isBuffer(input)) {
     return "At byte " + index.offset;
   }
-  var NEWLINE_REGEX = /\r\n|[\n\r\u2028\u2029]/;
 
-  var arrowUp = "^";
-  var arrowRight = ">";
-
-  var line = error.index.line;
-  var column = error.index.column;
-  var lines = map(function(lineSource, lineNumber) {
-    return {
-      lineNumber: lineNumber + 1,
-      lineSource: lineSource
-    };
-  }, input.split(NEWLINE_REGEX));
-  var lineNumberLabelLength = lines.length.toString().length;
-  var errorHighlight = {
-    lineNumber: repeat(" ", lineNumberLabelLength),
-    lineSource: repeat(" ", column - 1) + arrowUp
-  };
-  var linesWithErrorHighLight = [].concat(
-    lines.slice(0, line),
-    [errorHighlight],
-    lines.slice(line)
+  var line = index.line;
+  var column = index.column;
+
+  var inputLines = input.split(/\r\n|[\n\r\u2028\u2029]/);
+  var inputLinesLength = inputLines.length;
+  var lineWithErrorIndex = line - 1;
+  // Guard against the negative upper bound for lines included in the output.
+  var showFromLineIndex =
+    lineWithErrorIndex - 2 > 0 ? lineWithErrorIndex - 2 : 0;
+  var showToLineIndex =
+    lineWithErrorIndex + 3 > inputLinesLength
+      ? inputLinesLength
+      : lineWithErrorIndex + 3;
```

Comment:
Are `-2` and `+3` the number of lines of context to show before and after the error message? I think putting them in variables would help make it more clear what's going on here.

Reply:
Good point, will do!